manim/mobject/opengl/opengl_mobject.py

```diff
@@ -103,6 +103,59 @@ class MobjectStatus:
     points_changed: bool = False
 
 
+_O = TypeVar("_O")
+
+
+def affects_color(func) -> Callable[..., _O]:
+    """Decorator for methods that change the color of a :class:`~.Mobject`."""
+
+    def wrapper(self: OpenGLMobject, *args) -> _O:
+        self.status.color_changed = True
+        return func(self, *args)
+
+    return wrapper
+
+
+def affects_position(func) -> Callable[..., _O]:
+    """Decorator for methods that change the position of a :class:`~.Mobject`."""
+
+    def wrapper(self: OpenGLMobject, *args) -> _O:
+        self.status.position_changed = True
+        return func(self, *args)
+
+    return wrapper
+
+
+def affects_rotation(func) -> Callable[..., _O]:
+    """Decorator for methods that change the rotation of a :class:`~.Mobject`."""
+
+    def wrapper(self: OpenGLMobject, *args) -> _O:
+        self.status.rotation_changed = True
+        return func(self, *args)
+
+    return wrapper
+
+
+def affects_scale(func) -> Callable[..., _O]:
+    """Decorator for methods that change the scale of a :class:`~.Mobject`."""
+
+    def wrapper(self: OpenGLMobject, *args) -> _O:
+        self.status.scale_changed = True
+        return func(self, *args)
+
+    return wrapper
+
+
+def affects_points(func) -> Callable[..., _O]:
+    """Decorator for methods that change the points of a :class:`~.Mobject`."""
+
+    def wrapper(self: OpenGLMobject, *args) -> _O:
+        self.status.points_changed = True
+        return func(self, *args)
+
+    return wrapper
```

Comment:
Maybe use `functools.wraps` to keep function signature and documentation?

```suggestion
def affects_color(func) -> Callable[..., _O]:
    """Decorator for methods that change the color of a :class:`~.Mobject`."""

    @wraps(func)
    def wrapper(self: OpenGLMobject, *args) -> _O:
        self.status.color_changed = True
        return func(self, *args)

    return wrapper


def affects_position(func) -> Callable[..., _O]:
    """Decorator for methods that change the position of a :class:`~.Mobject`."""

    @wraps(func)
    def wrapper(self: OpenGLMobject, *args) -> _O:
        self.status.position_changed = True
        return func(self, *args)

    return wrapper


def affects_rotation(func) -> Callable[..., _O]:
    """Decorator for methods that change the rotation of a :class:`~.Mobject`."""

    @wraps(func)
    def wrapper(self: OpenGLMobject, *args) -> _O:
        self.status.rotation_changed = True
        return func(self, *args)

    return wrapper


def affects_scale(func) -> Callable[..., _O]:
    """Decorator for methods that change the scale of a :class:`~.Mobject`."""

    @wraps(func)
    def wrapper(self: OpenGLMobject, *args) -> _O:
        self.status.scale_changed = True
        return func(self, *args)

    return wrapper


def affects_points(func) -> Callable[..., _O]:
    """Decorator for methods that change the points of a :class:`~.Mobject`."""

    @wraps(func)
    def wrapper(self: OpenGLMobject, *args) -> _O:
        self.status.points_changed = True
        return func(self, *args)

    return wrapper
```

Reply:
Ah, sure! Yeah, I didn't think about documentation by now, but that seems like a good idea!